x/rollapp/types/genesis_info.go

```diff
@@ -53,35 +71,61 @@ func (gi GenesisInfo) Validate() error {
 
 	if gi.NativeDenom.IsSet() {
 		if err := gi.NativeDenom.Validate(); err != nil {
-			return errorsmod.Wrap(ErrInvalidNativeDenom, err.Error())
+			return errors.Join(ErrInvalidMetadata, err)
+		}
+	} else {
+		// if native denom is not set, initial supply must be 0 and no accounts
+		if !gi.InitialSupply.IsNil() && !gi.InitialSupply.IsZero() {
+			return ErrNoNativeTokenRollapp
```

Comment:
seems that the error here is not really the lack of native denom, but the fact that initial supply is non-zero while having no native tokens.